docs/recipes/reducers/00-PrerequisiteConcepts.md

```diff
@@ -0,0 +1,88 @@
+
+
+Because of these rules, it's important that the following core concepts are fully understood before moving on to other specific techniques for organizing Redux reducers:
+
+#### Redux Reducer Basics
+
+**Key concepts**:
+
+- Thinking in terms of state and state shape
+- Delegating update responsibility by slice of state (*reducer composition*)
+- Higher order reducers
+- Defining reducer initial state
+
+**Reading list**:
+
+- [Redux Docs: Reducers](../../basics/Reducers.md)
+- [Redux Docs: Reducing Boilerplate](../ReducingBoilerplate.md)
+- [Redux Docs: Implementing Undo History](../ImplementingUndoHistory.md)
+- [Redux Docs: `combineReducers`](../../api/combineReducers.md)
+- [The Power of Higher-Order Reducers](http://slides.com/omnidan/hor#/)
+- [Stack Overflow: Store initial state and `combineReducers`](http://stackoverflow.com/questions/33749759/read-stores-initial-state-in-redux-reducer)
+- [Stack Overflow: State key names and `combineReducers`](http://stackoverflow.com/questions/35667775/state-in-redux-react-app-has-a-property-with-the-name-of-the-reducer)
+
+
+#### Pure Functions and Side Effects
+
+**Key Concepts**:  
+
+- What "side effects" are
+- What "pure functions" are
+- How to think in terms of combining functions
+
+**Reading List**:
+
+- [The Little Idea of Functional Programming](http://jaysoo.ca/2016/01/13/functional-programming-little-ideas/)
+- [Understanding Programmatic Side Effects](http://web24studios.com/2015/10/understanding-programmatic-side-effects/)
+- [Learning Functional Programming in Javascript](https://youtu.be/e-5obm1G_FY)
+- [An Introduction to Reasonably Pure Functional Programming](https://www.sitepoint.com/an-introduction-to-reasonably-pure-functional-programming/)
+
+
+
+#### Immutable Data Management
+
+**Key Concepts**:
+
+- Mutability vs immutability
+- Immutably updating objects and arrays safely
+- Avoiding functions and statements that mutate state
+
+**Reading List**:
+
+- [Pros and Cons of Using Immutability With React](http://reactkungfu.com/2015/08/pros-and-cons-of-using-immutability-with-react-js/)
+- [Javascript and Immutability](http://t4d.io/javascript-and-immutability/)
+- [Immutable Data using ES6 and Beyond](http://wecodetheweb.com/2016/02/12/immutable-javascript-using-es6-and-beyond/)
+- [Immutable Data from Scratch](https://ryanfunduk.com/articles/immutable-data-from-scratch/)
+- [Redux Docs: Using the Object Spread Operator](../UsingObjectSpreadOperator.md)
+
+
+#### Normalizing Data
+
+**Key Concepts**:
+
+- Splitting relational/nested data up into separate tables
```

Comment:
Yeah, my first attempt at writing a couple of these sections, I found myself totally trying to explain concepts like "immutable data" from the ground up, and realized I was totally wasting effort re-explaining stuff that was better written and explained elsewhere.  I did do some explanation of normalization concepts and terms in those pages, but didn't go fully in depth.

Any suggestions for better handling the "prereqs" aspect?  Some way to better tie it in to the rest of the writing?